### ISSUE TITLE
[BUG] locator returns content that was removed from the page

### ISSUE THREAD

**Issue by benmccann** [microsoft/playwright#17557]
**Context:** browser tests for the SvelteKit framework
- Playwright Version: 1.25.0
- Operating System: Mac
- Node.js version: 16
- Browser: Webkit
- Extra: GitHub Actions

**Code Snippet**

Here's the failing test:

```javascript
test('+server.js next to +page.svelte works', async ({ page }) => {
	await page.goto('/routing/endpoint-next-to-page');
	expect(await page.textContent('p')).toBe('Hi');

	for (const method of ['GET', 'PUT', 'PATCH', 'POST', 'DELETE']) {
		await page.click(`button:has-text("${method}")`);
		await expect(page.locator('pre')).toHaveText(method);
	}
});
```

This issue occurs in a flaky manner. I've created a [standalone reproduction](https://github.com/benmccann/playwright-bug), but it's not triggering there - I expect because running a single test does not place the same load on the system as running many tests in parallel does. The issue happens relatively frequently when running SvelteKit's entire test suite. I've primarily seen it on Mac + Webkit. It's unclear to me if it happens elsewhere

**Describe the bug**

[This particular test](https://github.com/benmccann/kit/blob/7088999d341ac1ece1ff102d63dc46c15c19a69f/packages/kit/test/apps/basics/test/client.test.js#L903) clicks a button which [updates the page](https://github.com/benmccann/kit/blob/7088999d341ac1ece1ff102d63dc46c15c19a69f/packages/kit/test/apps/basics/src/routes/routing/endpoint-next-to-page/%2Bpage.svelte)'s `pre` element to say `loading`, makes a network call, and puts the result in the in the `pre` element. When running it on Github Actions `await expect(page.locator('pre')).toHaveText(method)` will frequently timeout saying that the text says `loading` eventhough the page has been updated an no longer contains that content.

Here's the output of the trace viewer from a run on the SvelteKit CI showing the bug:

![Screenshot from 2022-09-23 08-14-46](https://user-images.githubusercontent.com/322311/192027570-b341291d-019d-4a47-bfea-812dc5499686.png)

In the screenshot, you can see the highlighted text in the middle pane says `GET` and `loading` is nowhere to be found on the page. In the right pane you can see the log repeatedly says `unexpected value "loading"` as it's polling for the content of the `pre` element

Here are the CI logs from a job that experienced the flakiness:
```
@sveltejs/kit:test:   1) [webkit-build] › test/client.test.js:903:1 › +server.js next to +page.svelte works ============
@sveltejs/kit:test: 
@sveltejs/kit:test:     Error: expect(received).toHaveText(expected)
@sveltejs/kit:test: 
@sveltejs/kit:test:     Expected string: "GET"
@sveltejs/kit:test:     Received string: "loading"
@sveltejs/kit:test:     Call log:
@sveltejs/kit:test:       - expect.toHaveText with timeout 5000ms
@sveltejs/kit:test:       - waiting for selector "pre"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test: 
@sveltejs/kit:test: 
@sveltejs/kit:test:       907 | 	for (const method of ['GET', 'PUT', 'PATCH', 'POST', 'DELETE']) {
@sveltejs/kit:test:       908 | 		await page.click(`button:has-text("${method}")`);
@sveltejs/kit:test:     > 909 | 		await expect(page.locator('pre')).toHaveText(method);
@sveltejs/kit:test:           | 		                                  ^
@sveltejs/kit:test:       910 | 	}
@sveltejs/kit:test:       911 | });
@sveltejs/kit:test:       912 |
@sveltejs/kit:test: 
@sveltejs/kit:test:         at file:///Users/runner/work/kit/kit/packages/kit/test/apps/basics/test/client.test.js:909:37
@sveltejs/kit:test: 
@sveltejs/kit:test:     attachment #1: trace (application/zip) ---------------------------------------------------------
@sveltejs/kit:test:     test-results/test-client--server-js-next-to-page-svelte-works-webkit-build/trace.zip
@sveltejs/kit:test:     Usage:
@sveltejs/kit:test: 
@sveltejs/kit:test:         npx playwright show-trace test-results/test-client--server-js-next-to-page-svelte-works-webkit-build/trace.zip
@sveltejs/kit:test: 
@sveltejs/kit:test:     ------------------------------------------------------------------------------------------------
@sveltejs/kit:test: 
@sveltejs/kit:test:     attachment #2: screenshot (image/png) ----------------------------------------------------------
@sveltejs/kit:test:     test-results/test-client--server-js-next-to-page-svelte-works-webkit-build/test-failed-1.png
@sveltejs/kit:test:     ------------------------------------------------------------------------------------------------
@sveltejs/kit:test: 
@sveltejs/kit:test:     Retry #1 ---------------------------------------------------------------------------------------
@sveltejs/kit:test: 
@sveltejs/kit:test:     Error: expect(received).toHaveText(expected)
@sveltejs/kit:test: 
@sveltejs/kit:test:     Expected string: "GET"
@sveltejs/kit:test:     Received string: "loading"
@sveltejs/kit:test:     Call log:
@sveltejs/kit:test:       - expect.toHaveText with timeout 5000ms
@sveltejs/kit:test:       - waiting for selector "pre"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test:       -   selector resolved to <pre>loading</pre>
@sveltejs/kit:test:       -   unexpected value "loading"
@sveltejs/kit:test: 
@sveltejs/kit:test: 
@sveltejs/kit:test:       907 | 	for (const method of ['GET', 'PUT', 'PATCH', 'POST', 'DELETE']) {
@sveltejs/kit:test:       908 | 		await page.click(`button:has-text("${method}")`);
@sveltejs/kit:test:     > 909 | 		await expect(page.locator('pre')).toHaveText(method);
@sveltejs/kit:test:           | 		                                  ^
@sveltejs/kit:test:       910 | 	}
@sveltejs/kit:test:       911 | });
@sveltejs/kit:test:       912 |
@sveltejs/kit:test: 
@sveltejs/kit:test:         at file:///Users/runner/work/kit/kit/packages/kit/test/apps/basics/test/client.test.js:909:37
@sveltejs/kit:test: 
@sveltejs/kit:test:     attachment #1: trace (application/zip) ---------------------------------------------------------
@sveltejs/kit:test:     test-results/test-client--server-js-next-to-page-svelte-works-webkit-build-retry1/trace.zip
@sveltejs/kit:test:     Usage:
@sveltejs/kit:test: 
@sveltejs/kit:test:         npx playwright show-trace test-results/test-client--server-js-next-to-page-svelte-works-webkit-build-retry1/trace.zip
@sveltejs/kit:test: 
@sveltejs/kit:test:     ------------------------------------------------------------------------------------------------
@sveltejs/kit:test: 
@sveltejs/kit:test:     attachment #2: screenshot (image/png) ----------------------------------------------------------
@sveltejs/kit:test:     test-results/test-client--server-js-next-to-page-svelte-works-webkit-build-retry1/test-failed-1.png
@sveltejs/kit:test:     ------------------------------------------------------------------------------------------------
@sveltejs/kit:test: 
@sveltejs/kit:test: 
@sveltejs/kit:test:   1 flaky
@sveltejs/kit:test:     [webkit-build] › test/client.test.js:903:1 › +server.js next to +page.svelte works =============
@sveltejs/kit:test:   118 skipped
@sveltejs/kit:test:   387 passed (6m)
```

**Comment by aslushnikov** [comment]
I'll close this as per above. Unfortunately, this is not actionable without as being able to reproduce the issue!

Please do not hesitate to file a new issue to reach out to us with questions.
